General/Refactorings.md

```diff
@@ -0,0 +1,651 @@
+# Refactoring in Pharo 8.0
+
+## Introduction
+
+Pharo offers many automatic refactoring tools. Therefore, the objective of this tutorial is to present what refactorings are, when to apply them, how to apply them and to show any possible inconvenience before using them.
+
+## What is refactoring?
+
+Martin Fowler defines refactoring as: "... It is a disciplined technique to restructure an existing body of code, altering its internal structure without changing its external behavior." This means that before carrying out any type of refactoring in the code it is important to have complete test coverage to demonstrate that the code behavior does not change during the refactoring process.
+
+## The most popular refactorings supported by Pharo 8.0
+
+| Refactoring | Supported in |
+| ---------- | ---------- |
+| Rename   | Package, Class, Method, Instance and Temporary Variable |
+|  Extract   | Method and Temporary Variable  |
+| Remove| Package, Class, Method and Instance Variable |
+| Copy | Class |
+| Move | Class and method |
+| Inline | Method and Temporary Variable |
+| Push up / Push down | Method and Instance Variable |
+| Replace | Method |
+| Other refactorings|  Class, Method, Instance and Temporary Variable |
+
+# Rename
+
+Renaming seems like a trivial refactoring however it can lead to many complications because this action can lead to involuntary changes of elements not related to the same name. The Pharo's rename tool has can minimize these errors.
+
+## When should we rename?
+
+1. When the name is not descriptive enough
+2. The name of the class / method / variable does not match what it really is.
+3. Something new was introduced that requires the existing code to have a new more specific name.
+ 
+> This refactoring can be done for classes, methods and variables.
+>
+>1. Select the class / variable / method 
+>2. Press **⌘R** or from the menu, select **Rename**.
+>3. Fill the input of dialog with the new name and press **Ok** button.
+
+[![Pharo - Rename refactoring](https://img.youtube.com/vi/tHdt11WDq2o/0.jpg)](http://www.youtube.com/watch?v=tHdt11WDq2o)
+
+# Extract
+
+The extraction Pharo's tools allow developers to reshape their code when they have duplicate code or when they want to change the design.
+
+Pharo can perform this refactoring to extract methods and extract temporal variables.
+
+## Extract temporary variable
+
+<table>
```

Comment:
I think we should avoid to insert HTML code in the README as it reduces possible target renderer.

Thus, I would say that you should find a way to display this in another way to avoid HTML code.